### PR TITLE
NIAD-284: RECEP responder

### PR DESCRIPTION
**Pull request by grzegorz-robak**

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/configuration/NhaisMongoClientConfiguration.java [hunk 1]
```diff
@@ -61,10 +61,10 @@ protected boolean autoIndexCreation() {
     private String createConnectionString() {
         LOGGER.info("Creating a connection string for mongo client settings...");
         if (!Strings.isNullOrEmpty(host)) {
-            LOGGER.info("A value was provided from mongodb host. Generating a connection string from individual properties.");
+            LOGGER.info("A value was provided from mongodb host ({}). Generating a connection string from individual properties.", host);
             return createConnectionStringFromProperties();
         } else if (!Strings.isNullOrEmpty(uri)) {
-            LOGGER.info("A mongodb connection string provided in spring.data.mongodb.uri and will be used to configure the database connection.");
+            LOGGER.info("A mongodb connection string provided in spring.data.mongodb.uri ({}) and will be used to configure the database connection.", uri);
```
Comment (mattd-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/configuration/NhaisMongoClientConfiguration.java]: The connection string might contain a password. We should not log it.

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
Shouldn't log mongo URI - this may contain credentials

File: docker-compose.yml [hunk 2]
```diff
@@ -99,8 +99,13 @@ services:
           +yLvICQmVjHWsNWD3uuT3gEGdeKDIzEnovXUmqhQE7N/YDupTv5hkh8hMKl12XGs
           3WJ4bVSmp3MOQ2EF4zrAl4fXEd4lS3JxKtzbHFbSogAmPFnuP8wIDWTDgfEq
           -----END RSA PRIVATE KEY-----
-      - NHAIS_MESH_CYPHER_TO_MAILBOX=XX11=nhais_mailbox
-          TES5=nhais_mailbox
+      - NHAIS_MESH_CYPHER_TO_MAILBOX=XX11=responder_mailbox
```
Comment (mattd-kainos) [inline docker-compose.yml]: I'd stick with nhais_mailbox here - more clear that this is the mailbox we use to send messages to NHAIS

File: docker-compose.yml [hunk 3]
```diff
@@ -122,4 +127,34 @@ services:
...
+      - NHAIS_OUTBOUND_SERVER_PORT=8080
+      - NHAIS_AMQP_RETRY_DELAY=100
+      - NHAIS_AMQP_MAX_RETRIES=3
+      - NHAIS_AMQP_BROKERS=amqp://activemq:5672
+      - NHAIS_MESH_INBOUND_QUEUE_NAME=nhais_responder_mesh_inbound
+      - NHAIS_GP_SYSTEM_INBOUND_QUEUE_NAME=nahis_responder_gp_system_inbound
+      - NHAIS_MESH_OUTBOUND_QUEUE_NAME=nhais_responder_mesh_outbound
+      - NHAIS_MONGO_URI=mongodb://mongodb:27017
+      - NHAIS_MONGO_DATABASE_NAME=nhais_responder
+      - NHAIS_MESH_MAILBOX_ID=responder_mailbox
```
Comment (mattd-kainos) [inline docker-compose.yml]: And nhais_mailbox here

File: docker-compose.yml [hunk 4]
```diff
@@ -100,7 +100,12 @@ services:
           3WJ4bVSmp3MOQ2EF4zrAl4fXEd4lS3JxKtzbHFbSogAmPFnuP8wIDWTDgfEq
           -----END RSA PRIVATE KEY-----
       - NHAIS_MESH_CYPHER_TO_MAILBOX=XX11=nhais_mailbox
-          TES5=nhais_mailbox
+        TES5=nhais_mailbox
+        YY21=nhais_mailbox
+        KC01=nhais_mailbox
+        WHI5=nhais_mailbox
+        SO01=nhais_mailbox
+        UNY5=nhais_mailbox
```
Comment (Alicja-Kainos) [inline docker-compose.yml:108]: Isn't there lack of braces? Or the braces are not necessary? Doest it work properly?

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshService.java [hunk 5]
```diff
@@ -63,8 +69,9 @@ public void scanMeshInboxForMessages() {
             }
             LOGGER.info("Completed MESH mailbox polling cycle");
         } else {
-            LOGGER.info("Could not obtain database lock to run MESH mailbox polling cycle: insufficient time as elapsed " +
-                "since the previous polling cycle or another adaptor instance has already started the polling cycle.");
+            LOGGER.info("Could not obtain database lock to run MESH mailbox polling cycle: insufficient time has elapsed " +
+                "since the previous polling cycle or another adaptor instance has already started the polling cycle. " +
+                "Next scan in {} seconds", TimeUnit.SECONDS.convert(scanIntervalInMilliseconds, TimeUnit.MILLISECONDS));
```
Comment (Alicja-Kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshService.java:74]: I think next scan wont be in interval, but in interval minus the time from last reading. Although I am not sure if we can retrieve it such accurately. So maybe to say: 'Next scan not later than in {} seconds"?

File: src/recepResponder/java/uk/nhs/digital/nhsconnect/nhais/ResponderNhaisApplication.java [hunk 6]
```diff
@@ -21,7 +21,7 @@ public static void main(String[] args) {
 
 		@Bean
 		String registrationConsumerService() {
-			return RecepResponderService.class.getSimpleName();
+			return uk.nhs.digital.nhsconnect.nhais.responder.RecepResponderService.class.getSimpleName();
```
Comment (Alicja-Kainos) [inline src/recepResponder/java/uk/nhs/digital/nhsconnect/nhais/ResponderNhaisApplication.java]: why here is the whole path? Why not to import it like it was before?

File: docker-compose.yml [hunk 7]
```diff
@@ -122,4 +127,34 @@ services:
...
+      - NHAIS_MONGO_DATABASE_NAME=nhais_responder
+      - NHAIS_MESH_MAILBOX_ID=nhais_mailbox
+      - NHAIS_MESH_HOST=https://fake-mesh:8829/messageexchange/
+      - NHAIS_MESH_CYPHER_TO_MAILBOX=XX11=gp_mailbox
+        TES5=gp_mailbox
+        YY21=gp_mailbox
+        KC01=gp_mailbox
+        WHI5=gp_mailbox
+        SO01=gp_mailbox
+        UNY5=gp_mailbox
```
Comment (Alicja-Kainos) [inline docker-compose.yml:157]: lack of braces?

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshService.java (hunk 5 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshService.java:74]: Attempts to scan mailbox (obtain db lock) repeat in intervals, but actual scanning can be done after "delay time" has passed. By default it scans every 60 seconds to check if 5 minutes has passed from previous mailbox access.

File: src/recepResponder/java/uk/nhs/digital/nhsconnect/nhais/ResponderNhaisApplication.java (hunk 6 above)
Comment (grzegorz-robak) [reply src/recepResponder/java/uk/nhs/digital/nhsconnect/nhais/ResponderNhaisApplication.java]: It's IntelliJ refactor after I moved `ResponderNhaisApplication` from `uk.nhs.digital.nhsconnect.nhais.responder` to `uk.nhs.digital.nhsconnect.nhais`

File: docker-compose.yml (hunk 4 above)
Comment (grzegorz-robak) [reply docker-compose.yml:108]: It's `docker-compose.yml` file. You've mistaken it with `application.yml`

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshService.java (hunk 5 above)
Comment (Alicja-Kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshService.java:74]: All right, so the scan is not reading of the mailbox but just checking if reading can be proceeded? If so, I am ok with such message.

**Review by Alicja-Kainos — APPROVED** [review]

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**